lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -531,6 +531,13 @@ def _validate_final(self):
             # only search within the first collection we found
             self._subpackage_search_paths = [self._subpackage_search_paths[0]]
 
+    def exec_module(self, module):
+        collection_name = '.'.join(self._split_name[1:3])
+        AnsibleCollectionConfig.on_collection_load.fire(collection_name=collection_name, collection_path=os.path.dirname(module.__file__))
+
+    def create_module(self, spec):
+        return self.load_module(spec.name)
```

Comment:
This probably needs to change slightly, to perhaps not rely on `load_module`, however, maybe some of the code from `load_module` could move here, and then `load_module` could depend on `create_module`.

To explain a little bit, `pep489` is for a multi stage import system.  Where the stages are effectively:

1. `create_spec` (and loader)
2. `create_module` - this just creates the module container, but doesn't exec any code
3. `exec_module` - this execs the code

As it stands, `load_module` has historically done 2 and 3, and in the current implementation still calls `exec`

If you look at `importlib.module_from_spec` and `_new_module`, you will see the empty container is created by doing `type(sys)(spec.name)` effectively.

So `_new_or_existing_module` does too much for the context of `create_module`.